frmts/heif/heifdataset.cpp

```diff
@@ -700,10 +748,45 @@ void GDALRegister_HEIF()
     if (GDALGetDriverByName(DRIVER_NAME) != nullptr)
         return;
 
-    GDALDriver *poDriver = new GDALDriver();
-    HEIFDriverSetCommonMetadata(poDriver);
+    auto poDM = GetGDALDriverManager();
+    {
+        GDALDriver *poDriver = new GDALDriver();
+        HEIFDriverSetCommonMetadata(poDriver);
+
+#if LIBHEIF_NUMERIC_VERSION >= BUILD_LIBHEIF_VERSION(1, 12, 0)
+        // If the AVIF dedicate driver is not available, register a AVIF driver,
+        // called AVIF_HEIF, based on libheif, if it has AV1 decoding capabilities.
+        if (heif_have_decoder_for_format(heif_compression_AV1))
+        {
+            poDriver->SetMetadataItem("SUPPORTS_AVIF", "YES", "HEIF");
+        }
+#endif
 
-    poDriver->pfnOpen = GDALHEIFDataset::Open;
+        poDriver->pfnOpen = GDALHEIFDataset::OpenHEIF;
+        poDM->RegisterDriver(poDriver);
+    }
 
-    GetGDALDriverManager()->RegisterDriver(poDriver);
+#if LIBHEIF_NUMERIC_VERSION >= BUILD_LIBHEIF_VERSION(1, 12, 0)
+    // If the AVIF dedicate driver is not available, register a AVIF driver,
+    // called AVIF_HEIF, based on libheif, if it has AV1 decoding capabilities.
+    if (heif_have_decoder_for_format(heif_compression_AV1) &&
+        !poDM->IsKnownDriver("AVIF") && !poDM->IsKnownDriver("AVIF_HEIF"))
+    {
+        GDALDriver *poAVIF_HEIFDriver = new GDALDriver();
+        poAVIF_HEIFDriver->SetMetadataItem(GDAL_DCAP_RASTER, "YES");
+        poAVIF_HEIFDriver->SetDescription("AVIF_HEIF");
+        poAVIF_HEIFDriver->SetMetadataItem(
+            GDAL_DMD_LONGNAME, "AV1 Image File Format (using libheif)");
+        poAVIF_HEIFDriver->SetMetadataItem(GDAL_DMD_MIMETYPE, "image/avif");
+        poAVIF_HEIFDriver->SetMetadataItem(GDAL_DMD_HELPTOPIC,
+                                           "drivers/raster/avif.html");
```

Comment:
Should this point to the heif documentation instead?

Reply:
fixed